### PR TITLE
Adds resource requests and limits for api, db and ui

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: config/02-api/22-api-deployment.yaml [hunk 1]
```diff
@@ -41,6 +41,13 @@ spec:
       containers:
         - name: tekton-hub-api
           image: quay.io/tekton-hub/api
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+            limits:
+              cpu: 500m
+              memory: 500Mi
```
Comment (vinamra28) [inline config/02-api/22-api-deployment.yaml:50]: are these sufficient?

Comment (sm43) [reply config/02-api/22-api-deployment.yaml:50]: on what basis are we providing limits?
is it even required?

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Pull request merged**